### ISSUE TITLE
non tango phone data: irregular trajectory

### ISSUE THREAD

**Issue by arpitg1304** [higerra/ridi_imu#10]
I used your app: https://github.com/higerra/AndroidIMURecorder to collect data from my phone and then preprocessed using the python script. When I ran ./IMULocalization_cli for this data, I am not getting a regular trajectory. It shows movements in all three dimensions unlike all of the trajectories generated on your data which are flat
image
![image](https://user-images.githubusercontent.com/16556719/56183691-7fee8b80-5fe5-11e9-94b8-ed5f69ec3f2c.png)

Do you have any idea why this might be happening?

**Comment by Ador2** [comment]
what type is your phone?

**Comment by arpitg1304** [comment]
I tried with two phones:
-Samsung Galaxy S9
-Honor 6X

I got 3D trajectories in both cases(Unlike the 2D trajectories generated for the dataset provided with RIDI). Have you used RIDI with non-tango phone?

**Comment by higerra** [comment]
Hi,
We tested on the Lenovo Phab Pro 2 and Google Pixel XL.
We simply ignore the 3rd axis to get a 2D trajectory.

**Comment by arpitg1304** [comment]
Ok, are the trajectories accurate enough to get the distance traveled?

**Comment by higerra** [comment]
It largely depends on the training data and motions. Since the number of people that train the model is limited, I'm not expecting high accuracy with pretrained model.

**Comment by Ador2** [comment]
Well Honor 6X doesn't have a gyro, and as @higerra said it needs to train well on large amount of training data, I've tried the algorithm on a non-tango device and it was showing a little bit movement on the 3rd axis and that completely fine

**Comment by arpitg1304** [comment]
Honor 6X has a gyro and I also tested with Galaxy S9. Even if the trajectory is in 2D, the distance traveled (norm of poses) is very inaccurate. I held my phone and walked for about 11 meters while recording the data. 

Then, I ran ./IMULocalization_cli on that data. The norm between first and last pose is only 4.5 meters. 

@higerra Is the pre-trained model different than the one you used for generating results in your videos and paper?

**Comment by higerra** [comment]
It's the same, but trained specifically for one person. We were using a traditional machine learning model (SVR), so it might not do well on large dataset.

If you understand the paper are doing research on this, I would recommend trying:
1. Ditch the "stabilized-IMU frame" formulation. We found later that this formulation has some singularity issue. Instead, simply regress velocities on 3 local axes.
2. Use a deep learning based model, for example, ResNet, or LSTM.
You can still use a large portion of RIDI codebase as a starting point of your implementation. It should work well on provided dataset.

Then, if you have the resource, you can try collecting a lot more training data. This is very effective with deep learning models.
